### PR TITLE
update authentication docs to point to tsec-http4s

### PR DESCRIPTION
**Pull request by jmcardon**
I wasn't sure whether tsec-auth should be at the top, or at the bottom, so I put it at the bottom by default.

If you guys feel tsec-http4s is good, we can move it accordingly so it's easier to see.

### CODE REVIEW

File: docs/src/main/tut/auth.md [hunk 1]
```diff
@@ -132,6 +134,12 @@ val authUser: Kleisli[IO, Request[IO], Either[String,User]] = Kleisli({ request
 })
 ```
 
+### Using tsec-http4s for Authentication and Authorization
+The [TSec] project provides an authentication and authorization module
+ for the http4s project 0.18-M4+. Docs specific to http4s are located [here].
```
Comment (ChristopherDavenport) [inline docs/src/main/tut/auth.md]: If your going to use a world like `here` I think the link belongs inline.

**Comment by jmcardon** [comment]
Also, could this pr close #1085 ?

**Comment by jmcardon** [comment]
@ChristopherDavenport I fixed the definition of service in that paragraph in the last commit, but opened #1527 for the remaining, so it can be a bit more specific.

File: docs/src/main/tut/index.md [hunk 2]
```diff
@@ -4,6 +4,10 @@ menu: main
 weight: 1
 ---
 
+**Note**: To run examples, please make sure that the flag `-Ypartial-unification`
```
Comment (rossabaker) [inline docs/src/main/tut/index.md]: Should we also note that it's enabled by default by the g8 template?

Comment (jmcardon) [reply docs/src/main/tut/index.md]: Sure, I'm alright with this. I'm all for anything that makes it more obvious that it's enabled, as it's pretty much the most common source of errors currently.

Comment (jmcardon) [reply docs/src/main/tut/index.md]: updated.

**Pull request merged**